### ISSUE TITLE
Transparent Hugepages research

### ISSUE THREAD

**Issue by backstreetkiwi** [stackabletech/t2#348]
While fixing Hetzner network problems back in #334, it helped to remove the disabling of Transparent Hugepages (THP) in the kernel. We don't really know how Hetzner networking and Kernel options are related, but it was thoroughly tested that there was a causality somehow.

Originally, the THP disabling was copied over from the scripts from @soenkeliebau which were the origin of all the T2 stuff. So we should discuss with him if this is needed at all. (The nightly operator tests work fine with THP on, but maybe they don't reach the spheres where THPs make a difference. The managed K8s we use do not let us control this at all.)

If we have to take care of THP disabling, it would be a good idea to set up two Hetzner clusters with THPs en/disabled and further investigate it. Teaming up with @teozkr should be valuable here.

**Comment by soenkeliebau** [comment]
Transparent hugepages are or used to be a really important optimization for hdfs performance.
I'm not sure if there are new nest practices around this in recent years, @lfrancke @Jimvin do you have more knowledge here?

**Comment by lfrancke** [comment]
Yeah, HBase was the worst affected. I believe it is still a problem but if it causes other issues then we can live with lower performance.
That said....I don't really understand how THP has any effect on network stability but...

**Comment by Jimvin** [comment]
The side effect of enabling THP would typically be high system CPU. If this overhead was high enough it could impact other kernel operations, including networking. I would recommend disabling it, although the actual effect of THP is likely dependent on the workload and the kernel version. I cannot see an obvious advantage to keeping it enabled.

**Comment by backstreetkiwi** [comment]
THP does not affect network stability at runtime but the GRUB change interfered somehow with the Hetzner stuff that sets up the server/network. You are right, it sounds funny and the first thing in this task would be to investigate exactly that.

**Comment by lfrancke** [comment]
Ah! I see. THP can also be disabled outside of GRUB. It used to be done in an rc.local file but I admit it's been a while and I'd need to dig into the details as well.

**Comment by backstreetkiwi** [comment]
Thanks for your comments. During a restructuring I recently did in my Ansible roles I could once more see that there really is a connection between the THP settings (in the way I did them) and the Network Manager in Hetzner, although probably an indirect one. I'll try another way of disabling it outside of GRUB.

**Comment by backstreetkiwi** [comment]
This issue dates back to the days of Hetzner / K3s. I think we can drop this issue, @Jimvin
